core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -17,11 +21,13 @@
 /**
  * Encloses classes related to the compression and decompression of messages.
  */
+@ExperimentalApi
 public final class MessageEncoding {
   /**
    * Special sentinel codec indicating that no compression should be used.  Users should use
    * reference equality to see if compression is disabled.
    */
+  @ExperimentalApi
```

Comment:
The benefits that stand out to me to have this be a method:
- Nice semantic name 
- Easy to change out the returned object

The original reason I had not gone with this was that doing == on a constant makes sense to me, where as calling == on the return value of a method is more suspect.  Would you be surprised to see "if (decoder == Codec.identity())" ?